cpp/src/arrow/flight/flight_test.cc

```diff
@@ -1515,6 +1529,26 @@ TEST_F(TestDoPut, DoPutDicts) {
   CheckDoPut(descr, schema, batches);
 }
 
+// Ensure the gRPC server is configured to allow large messages
+// Tests a 32 MiB batch
+TEST_F(TestDoPut, DoPutLargeBatch) {
+  const auto array_length = 32768;
+  auto descr = FlightDescriptor::Path({"floats"});
+  BatchVector batches;
+  std::vector<std::shared_ptr<arrow::Array>> arrays;
```

Comment:
Can you delegate the record batch creation to a helper function, since it's done both for DoGet and DoPut tests?